Rename Android TV to Android Debug Bridge (#26830)
home-assistant/home-assistant.github.io: source/_integrations/androidtv.markdown

---

> title: Android debug bridge

Reviewer: https://developer.android.com/studio/command-line/adb
```suggestion
title: Android Debug Bridge
```

ha_integration_type: device
---

> The `androidtv` integration allows you to control an Android device or [Amazon Fire TV](https://www.amazon.com/b/?node=8521791011) device.

Reviewer: ```suggestion
The Android Debug Bridge integration allows you to control an Android device or [Amazon Fire TV](https://www.amazon.com/b/?node=8521791011) device.
```

### `androidtv.download` and `androidtv.upload`

You can use the `androidtv.download` service to download a file from your Android / Fire TV device to your Home Assistant instance.

| Service data attribute | Optional | Description |
| ---------------------- | -------- | ----------- |
| `entity_id`            |       no | Name of Android / Fire TV entity.
| `device_path`          |       no | The filepath on the Android / Fire TV device.
| `local_path`           |       no | The filepath on your Home Assistant instance.

Similarly, you can use the `androidtv.upload` service to upload a file from Home Assistant instance to Android / Fire TV devices.

| Service data attribute | Optional | Description |
| ---------------------- | -------- | ----------- |
| `entity_id`            |       no | Name(s) of Android / Fire TV entities.
| `device_path`          |       no | The filepath on the Android / Fire TV device.
| `local_path`           |       no | The filepath on your Home Assistant instance.

## Custom State Detection

> The Android debug bridge integration works by polling the Android / Fire TV device at a regular interval and collecting a handful of properties. Unfortunately, there is no standard API for determining the state of the device to which all apps adhere. Instead, the backend `androidtv` package uses three of the properties that it collects to determine the state: `audio_state`, `media_session_state`, and `wake_lock_size`. The correct logic for determining the state differs depending on the current app, and the backend `androidtv` package implements app-specific state detection logic for a handful of apps. Of course, it is not feasible to implement custom logic for each and every app in the `androidtv` package. Moreover, the correct state detection logic may differ across devices and device configurations.

Reviewer: ```suggestion
The Android Debug Bridge integration works by polling the Android / Fire TV device at a regular interval and collecting a handful of properties. Unfortunately, there is no standard API for determining the state of the device to which all apps adhere. Instead, the backend `androidtv` package uses three of the properties that it collects to determine the state: `audio_state`, `media_session_state`, and `wake_lock_size`. The correct logic for determining the state differs depending on the current app, and the backend `androidtv` package implements app-specific state detection logic for a handful of apps. Of course, it is not feasible to implement custom logic for each and every app in the `androidtv` package. Moreover, the correct state detection logic may differ across devices and device configurations.
```
